client/blocks/subscription-length-picker/index.jsx

```diff
@@ -0,0 +1,193 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal Dependencies
+ */
+import { getPlansBySiteId, isRequestingSitePlans } from 'state/sites/plans/selectors';
+import { localize } from 'i18n-calypso';
+import formatCurrency from 'lib/format-currency';
+import { getProductsList, isProductsListFetching } from 'state/products-list/selectors';
+import { requestProductsList } from 'state/products-list/actions';
+import { requestPlans } from 'state/plans/actions';
+import { getPlanRawPrice, isRequestingPlans } from '../../state/plans/selectors';
+import { getCurrentUserCurrencyCode } from '../../state/current-user/selectors';
+import { getPlanDiscountedRawPrice } from '../../state/sites/plans/selectors';
+import { getSelectedSiteId } from '../../state/ui/selectors';
+import { abtest } from 'lib/abtest';
+import { getPlan, applyTestFiltersToPlansList } from 'lib/plans';
+import { TERM_MONTHLY } from 'lib/plans/constants';
+import SubscriptionLengthOption from './option';
+
+export class SubscriptionLengthPicker extends React.Component {
+	static propTypes = {
+		initialValue: PropTypes.string,
+		currencyCode: PropTypes.string.isRequired,
+		productsWithPrices: PropTypes.array.isRequired,
```

Comment:
Also not sure why this is required, when we try to build this in mapStateToProps

Reply:
Because this component requires it to render, is there a reason why it shouldn't be marked as required?